src/DynamicRupture/Initializers/BaseDRInitializer.cpp

```diff
@@ -1,126 +1,242 @@
+    }
+  }
+}
+
+std::vector<unsigned>
+    BaseDRInitializer::getFaceIDsInIterator(seissol::initializers::DynamicRupture* dynRup,
+                                            seissol::initializers::LTSTree::leaf_iterator& it) {
+  const auto& drFaceInformation = it->var(dynRup->faceInformation);
+  std::vector<unsigned> faceIDs;
+  // collect all face IDs within this lts leaf
+  for (unsigned int ltsFace = 0; ltsFace < it->getNumberOfCells(); ++ltsFace) {
+    faceIDs.push_back(drFaceInformation[ltsFace].meshFace);
+  }
+  return faceIDs;
+}
 
-    } // lts-face loop
-    layerLtsFaceToMeshFace += it->getNumberOfCells();
-  } // leaf_iterator loop
+void BaseDRInitializer::queryModel(seissol::initializers::FaultParameterDB& faultParameterDB,
+                                   std::vector<unsigned> faceIDs) {
+  // create a query and evaluate the model
+  double boundaryGaussPoints[numPaddedPoints][2] = {0};
+  double weights[numPaddedPoints] = {0};
+  quadrature::TriangleQuadrature(boundaryGaussPoints, weights, CONVERGENCE_ORDER + 1);
+  seissol::initializers::FaultGPGenerator queryGen(
+      seissol::SeisSol::main.meshReader(),
+      reinterpret_cast<double(*)[2]>(boundaryGaussPoints),
+      numPaddedPoints,
+      faceIDs);
+  faultParameterDB.evaluateModel(drParameters.faultFileName, queryGen);
 }
+
+void BaseDRInitializer::rotateStressToFaultCS(seissol::initializers::DynamicRupture* dynRup,
+                                              seissol::initializers::LTSTree::leaf_iterator& it,
+                                              real (*stressInFaultCS)[numPaddedPoints][6],
+                                              real (*stressXX)[numPaddedPoints],
+                                              real (*stressYY)[numPaddedPoints],
+                                              real (*stressZZ)[numPaddedPoints],
+                                              real (*stressXY)[numPaddedPoints],
+                                              real (*stressYZ)[numPaddedPoints],
+                                              real (*stressXZ)[numPaddedPoints]) {
+  for (unsigned int ltsFace = 0; ltsFace < it->getNumberOfCells(); ++ltsFace) {
+    const auto& drFaceInformation = it->var(dynRup->faceInformation);
+    unsigned meshFace = static_cast<int>(drFaceInformation[ltsFace].meshFace);
+    const Fault& fault = seissol::SeisSol::main.meshReader().getFault().at(meshFace);
+    Eigen::Matrix<double, 6, 6> rotationMatrix;
+    seissol::transformations::inverseSymmetricTensor2RotationMatrix(
+        fault.normal, fault.tangent1, fault.tangent2, rotationMatrix, 0, 0);
+
+    for (unsigned int j = 0; j < numPaddedPoints; ++j) {
+      Eigen::Vector<double, 6> stress;
+      stress << stressXX[ltsFace][j], stressYY[ltsFace][j], stressZZ[ltsFace][j],
+          stressXY[ltsFace][j], stressYZ[ltsFace][j], stressXZ[ltsFace][j];
+      Eigen::Vector<double, 6> rotatedStress = rotationMatrix * stress;
+      for (int k = 0; k < 6; ++k) {
```

Comment:
k - stress or something 

maybe make `6` is a constant expression. We actually need `6` in more than one place in seissol. It make sense to make it a global constant expression or macro